### PR TITLE
[MAINTENANCE] Backfill checkpoint ID/PK integration tests

### PR DESCRIPTION
**Pull request by cdkini**
Write tests to cover ID/PK feature with new V1 checkpoint (replaces deleted tests)

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tests/datasource/fluent/integration/test_sql_datasources.py [hunk 1]
```diff
@@ -710,5 +711,131 @@ def _raw_query_check_column_exists(
...
+        param("QUOTED_LOWER_COL", id="str QUOTED_LOWER_COL"),
+        param('"QUOTED_LOWER_COL"', id='str "QUOTED_LOWER_COL"'),
+        # DDL: "QUOTED_UPPER_COL" ----------------------------------
+        param("quoted_upper_col", id="str quoted_upper_col"),
+        param('"quoted_upper_col"', id='str "quoted_upper_col"'),
+        param("QUOTED_UPPER_COL", id="str QUOTED_UPPER_COL"),
+        param('"QUOTED_UPPER_COL"', id='str "QUOTED_UPPER_COL"'),
+    ],
+)
+class TestColumnIdentifiers:
```
Comment (billdirks) [inline tests/datasource/fluent/integration/test_sql_datasources.py:741]: I don't believe this is an ID/PK test but I'm glad that this is back. Be had a lot of bugs around quoted identifiers and casing especially with snowflake.
If other tests in this file were deleted from this file, you may want to take another look to see if any others should come back. Happy to go through them with you.

File: tests/checkpoint/test_checkpoint_id_pk.py [hunk 2]
```diff
@@ -0,0 +1,557 @@
...
+    assert index_column_names == ["pk_1"]
+
+    first_result_full_list = evrs[0]["results"][0]["result"]["unexpected_index_list"]
+    assert first_result_full_list == expected_unexpected_indices_output
+
+    first_result_partial_list = evrs[0]["results"][0]["result"]["partial_unexpected_index_list"]
+    assert first_result_partial_list == expected_unexpected_indices_output
+
+    unexpected_index_query = evrs[0]["results"][0]["result"]["unexpected_index_query"]
+    assert unexpected_index_query == expected_sql_query_output
```
Comment (billdirks) [inline tests/checkpoint/test_checkpoint_id_pk.py:204]: Are the asserts here suppose to be identical to the previous test, `test_sql_result_format_in_checkpoint_pk_defined_one_expectation_complete_output`?

**Review by billdirks — COMMENTED** [review]
This LGTM. I asked @Shinnnyshinshin to give the reintroduced id/pk tests a look to help make sure we've covered the major workflows.

**Review by billdirks — APPROVED** [review]

File: tests/datasource/fluent/integration/test_sql_datasources.py (hunk 1 above)
Comment (Shinnnyshinshin) [reply tests/datasource/fluent/integration/test_sql_datasources.py:741]: agreed. Thank you for this @cdkini

File: tests/checkpoint/test_checkpoint_id_pk.py [hunk 3]
```diff
@@ -0,0 +1,557 @@
...
+    )
+
+
+@pytest.fixture
+def expect_column_pair_values_to_be_equal() -> gxe.ExpectColumnPairValuesToBeEqual:
+    return gxe.ExpectColumnPairValuesToBeEqual(column_A="ordered_item", column_B="received_item")
+
+
+@pytest.fixture
+def expect_multicolumn_sum_to_equal() -> gxe.ExpectMulticolumnSumToEqual:
```
Comment (Shinnnyshinshin) [inline tests/checkpoint/test_checkpoint_id_pk.py:100]: really appreciate how these fixtures were set up. Thank you @cdkini

**Review by Shinnnyshinshin — APPROVED** [review]
Thank you @cdkini this looks great. This is a non-blocking question, but has there been any discussion around simplifying the path around configuring `ResultFormat`?  (currently as either a enum or dict)

**Pull request merged**